core-web/libs/portlets/dot-content-drive/portlet/src/lib/components/dot-content-drive-toolbar/components/dot-content-drive-content-type-field/dot-content-drive-content-type-field.component.ts

```diff
@@ -0,0 +1,130 @@
+
+    // We need to map the numbers to the base types, ticket: https://github.com/dotCMS/core/issues/32991
+    // This prevents the effect from being triggered when the base types are the same or filters changes
+    private readonly $mappedBaseTypes = computed(
+        () => this.#store.filters().baseType?.map((item) => MAP_NUMBERS_TO_BASE_TYPES[item]) ?? [],
+        {
+            // This will trigger the effect if the base types are different
+            equal: (a, b) => a?.length === b?.length && a?.every((item) => b.includes(item))
+        }
+    );
+
+    constructor() {
+        // Set up debounced API request stream with switchMap
+        this.#apiRequestSubject
+            .pipe(
+                tap(() => patchState(this.$state, { loading: true, contentTypes: [] })),
+                debounceTime(DEBOUNCE_TIME),
+                distinctUntilChanged(
+                    (prev, curr) => prev.type === curr.type && prev.filter === curr.filter
+                ),
+                switchMap(({ type, filter }) =>
+                    this.#contentTypesService
+                        .getContentTypes({ type, filter })
+                        .pipe(catchError(() => of([])))
+                ),
+                takeUntilDestroyed()
+            )
+            .subscribe((contentTypes: DotCMSContentType[]) => {
+                const contentTypeFallback = contentTypes ?? [];
+
+                patchState(this.$state, {
+                    contentTypes: contentTypeFallback?.filter(
+                        (item) => item.baseType !== BASE_TYPES.form && !item.system
+                    ),
+                    loading: false
+                });
+
+                const contentTypeFilters = this.#store.getFilterValue('contentType');
+
+                this.$selectedContentTypes.set(
+                    contentTypeFallback.filter((item) =>
+                        contentTypeFilters?.includes(item.variable)
+                    )
+                );
+            });
+    }
+
+    readonly getContentTypesEffect = effect(() => {
+        // TODO: We need to improve this when this ticket is done: https://github.com/dotCMS/core/issues/32991
+        // After that remove the uncommented code and
+        // const type = this.$mappedBaseTypes().join(',')
+        const type = undefined;
+        const filter = this.$state.filterByKeyword();
+
+        // Push the request parameters to the debounced stream
+        this.#apiRequestSubject.next({ type, filter });
+    });
+
+    onFilter(event: MultiSelectFilterEvent) {
+        patchState(this.$state, {
+            filterByKeyword: event.filter
+        });
+    }
+
+    onChange() {
+        const value = this.$selectedContentTypes() ?? [];
+
+        if (value.length > 0) {
+            this.#store.patchFilters({
+                contentType: value.map((item) => item.variable)
+            });
+        } else {
+            this.#store.removeFilter('contentType');
```

Comment:
Same as before.

Reply:
We don't want to bloat the toolbar, because it will have more logic in the near future, but we haven't added anything there yet

Reply:
Also this component is from my [PR](https://github.com/dotCMS/core/pull/33013) it is not merged yet, but kev needed some changes from there in other to work with no problems